source/material_model/rheology/strain_dependent.cc

```diff
@@ -271,21 +271,32 @@ namespace aspect
                          Parameters<dim>::NonlinearSolver::single_Advection_iterated_Stokes
                          ||
                          this->get_parameters().nonlinear_solver ==
+                         Parameters<dim>::NonlinearSolver::iterated_Advection_and_Stokes
```

Comment:
Does this mean we're now checking against all the available solver schemes? Can the whole Assert be removed? Or switched to checking what solvers should _not_ be used?
Or is it better to keep the Assert for future cases where new solvers are introduced?

Reply:
Good point. My preference for now would be to keep the current assert in case something does break with the current allowed solvers. However, if everything seems to be working smoothly after some period of time (months, one year) then maybe we can remove it altogether?